### PR TITLE
Fro 45 knowledge library

### PR DESCRIPTION
**Pull request by ejim11**
# Fixes Issue/Linear Ticket

( This could be an existing issue or a linear ticket )

**My PR Closes #issue_number_here** or
**My PR Fixes ID-linear_ticket_number_here**

# Changes proposed

## What did you do?

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.
- [] I am making a pull request against the **dev branch** (left side).
- [] My commit messages styles matches our requested structure.
- [] My code additions will fail neither code linting checks nor unit test.
- [] I am only making changes to files I was requested to.

# Screenshots/ Videos

### CODE REVIEW

File: packages/main/src/pages/KnowledgeLibrary/Categories/Categories.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import navData from "../Navigation/NavData";
+import classes from "./Categories.module.css";
+import { Link } from "react-router-dom";
+
+function Categories(props) {
+  return (
+    <div className={classes["container"]}>
+      <p className={`${classes["header-p"]} ${props.className}`}>Categories</p>
+      {navData.map((item, i) => (
+        <div className={classes["item-box"]} key={i}>
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/Categories/Categories.jsx:10]: index is not a good key to map with

File: packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { useState } from "react";
+import classes from "./CategoryList.module.css";
+import rightLogo from "../assets/right.svg";
+import downLogo from "../assets/down.svg";
+
+const CategoryList = ({ item }) => {
+  const [displaySubCategory, setDisplaySubCategory] = useState("");
+
+  const toggleSubCategories = e => {
+    console.log(displaySubCategory);
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx:10]: don't log

File: packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if (displaySubCategory === e.target.dataset.sub) {
+      setDisplaySubCategory("");
+      console.log("now");
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx:13]: don't log

File: packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        data-sub={item.title}
+      />
+      {item.title}
+      <ul
+        className={
+          displaySubCategory === item.title ? classes["display"] : classes.hide
+        }
+      >
+        {item.subtitles?.map((listItem, i) => (
+          <li key={i} data-sub="sub-item">
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/Categories/CategoryList.jsx:40]: don't map with index
it defeats the point of a key

File: packages/main/src/pages/KnowledgeLibrary/KnowledgeLibrary.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        <main className={classes.main}>
+          <h2>Welcome to Zuri chat Knowledge library</h2>
+          <p className={classes["head-p"]}>
+            Find useful company information in one place
+          </p>
+          <Switch>
+            <Route path={`${path}`} component={() => <Home />} exact />
+            <Route
+              path={`${path}/categories`}
+              component={() => <Categories />}
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/KnowledgeLibrary.jsx:25]: why's it being called?

File: packages/main/src/pages/KnowledgeLibrary/Navigation/Navigation.jsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import categoriesLogo from "../assets/clipboard-text.svg";
+import downLogo from "../assets/down.svg";
+import rightLogo from "../assets/right.svg";
+import CategoryList from "../Categories/CategoryList";
+
+const Navigation = () => {
+  let { path } = useRouteMatch();
+  const location = useLocation();
+  const curPath = location.pathname.split("/").slice(-1)[0];
+  console.log(curPath);
```
Comment (ickynavigator) [inline packages/main/src/pages/KnowledgeLibrary/Navigation/Navigation.jsx:15]: do not log

**Review by ickynavigator — CHANGES_REQUESTED** [review]

**Comment by JulianaSau** [comment]
Please provide the details requested in your PR description and include relevant screenshots or videos @ejim11

File: packages/main/src/pages/KnowledgeLibrary/Categories/Categories.jsx (hunk 1 above)
Comment (ejim11) [reply packages/main/src/pages/KnowledgeLibrary/Categories/Categories.jsx:10]: done, i hope it's okay to use an obj as a key

Comment (JulianaSau) [reply packages/main/src/pages/KnowledgeLibrary/Categories/Categories.jsx:10]: Give an example of what you'd like to try to use

**Comment by JulianaSau** [comment]
And please deal with the conflicts😁

**Pull request closed without merging**